crates/aptos/src/node/local_testnet/indexer_api.rs

```diff
@@ -139,40 +149,14 @@ impl ServiceManager for IndexerApiManager {
             ..Default::default()
         };
 
-        let docker = get_docker().await?;
+        // Rather than try to connect to 127.0.0.1 from inside the container, connect
+        // to the hostname we just set up that resolves to host-gateway, which itself
+        // lets you access localhost on the host.
+        let postgres_connection_string = self
+            .postgres_connection_string
+            .replace("127.0.0.1", HOST_LOCALHOST_HOSTNAME);
 
-        // When using Docker Desktop you can and indeed must use the magic hostname
-        // host.docker.internal in order to access localhost on the host system from
-        // within the container. This also theoretically works without Docker Desktop,
-        // but you have to manually add the name to /etc/hosts in the container, and in
-        // my experience even that doesn't work sometimes. So when in a Docker Desktop
-        // environment we replace 127.0.0.1 with host.docker.internal, whereas in other
-        // environments we still use 127.0.0.1 and use host networking mode.
-        //
-        // In practice, this means we do the replacement when on MacOS or Windows, both
-        // standard (NT) and WSL and we don't do it on Linux / when running from within
-        // a container. But checking for OS is not accurate, since for example we must
-        // do the replacement when running in WSL configured to use the host Docker
-        // daemon but not when running in WSL configured to use Docker from within the
-        // WSL environment. So instead of checking for OS we check the name of the
-        // Docker daemon.
-        //
-        // This is the best method I could figure out for determining the Docker env,
-        // see more here: https://stackoverflow.com/q/77243960/3846032.
-        let info = docker
-            .info()
-            .await
-            .context("Failed to get info about Docker daemon")?;
-        let is_docker_desktop = info.operating_system == Some("Docker Desktop".to_string());
-        let postgres_connection_string = if is_docker_desktop {
-            info!("Running with Docker Desktop, using host.docker.internal");
-            self.postgres_connection_string
-                .replace("127.0.0.1", "host.docker.internal")
-        } else {
-            info!("Not running with Docker Desktop, using host networking mode");
-            host_config.network_mode = Some("host".to_string());
-            self.postgres_connection_string
-        };
+        let docker = get_docker().await?;
```

Comment:
yay, local testnet was broken on my windows machine earlier because of docker but this one fixes it 😄 

I did have to install `sudo apt-get install libdw-dev` but with the other one I couldn't even get it to work. so yay!

Reply:
Sadly this is actually no good, it doesn't work on Linux :/ 

Incredibly I keep researching this and I still can't find a way to connect to a service on the host from a container that works for every platform, I have to detect the platform somehow and change how I create the container.

Reply:
This sums up the issue: https://stackoverflow.com/questions/77431086/how-to-access-service-running-on-host-from-inside-docker-container-in-a-cross-pl. Hopefully someone else knows what to do here.

Reply:
Okay goddamn I actually figured it out finally. It works on MacOS and Linux, testing on Windows now.

Reply:
Bingo it works on Windows (WSL) too. Beautiful!!

Reply:
The fix at first was to change the address I was binding postgres to, but now I'm letting the containers communicate through a network I create with Docker (bridge network). That eliminates the need for containers to connect with each other through the host, which has different behavior in different environments.